test/presubmit-config-check.sh

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+source $(dirname $0)/../scripts/library.sh
+
+set -e
+
+trap 'echo "Please rerun \`make -C config/prow config\`."' ERR
+header "Checking generated config for production prow and testgrid"
+export PROW_CONFIG=$(mktemp)
+export PROW_JOB_CONFIG=$(mktemp)
+export PROW_PLUGINS=$(mktemp)
+export TESTGRID_CONFIG=$(mktemp)
```

Comment:
You aren't actually writing to these files. This bash script has to invert the logic of the Makefile; first copy each of the files to the temp file, run make config, then compare them.

Reply:
We are using `?=` in the Makefile, like https://github.com/knative/test-infra/blob/4dac123b9a3d14d2e07b4a59fbac0b0e99a09e92/config/prow_common.mk#L32.

If these env vars are defined as what we are doing here with `export XXX=xx`, the make command will actually use them. So it actually works :)

Reply:
I thought that only related to make variables and not environment variables. Or do all environment variables become make variables?
I don't like that the environment could make the command misbehave, but you're not making it worse. So if it works, it works.